lib/segment/src/entry/entry_point.rs

```diff
@@ -297,13 +306,35 @@ pub trait SegmentEntry: SnapshotEntry {
         field_schema: Option<&PayloadFieldSchema>,
         hw_counter: &HardwareCounterCell,
     ) -> OperationResult<bool> {
-        let Some((schema, index)) =
-            self.build_field_index(op_num, key, field_schema, hw_counter)?
-        else {
-            return Ok(false);
+        let Some(field_schema) = field_schema else {
+            // Legacy case, where we tried to automatically detect the schema for the field.
+            // We don't do this anymore, as it is not reliable.
+            return Err(OperationError::TypeInferenceError {
+                field_name: key.clone(),
+            });
         };
```

Comment:
Can't we enforce this at the interface level?

Reply:
That would be a breaking change for the interface.

Reply:
I think it could be done later in a different PR to clean things up :+1: